tests/unit/evaluation/SplittingStrategy_unittest.cc

```diff
@@ -296,3 +296,90 @@ TEST(SplittingStrategy,LOO)
 		SG_UNREF(splitting);
 	}
 }
+
+TEST(SplittingStrategy, timeseries_subset_linear_splits)
+{
+	index_t num_labels, num_subsets, min_subset_size, base_size;
+	index_t runs = 10;
+
+	while (runs-- > 0)
+	{
+		num_labels = CMath::random(50, 150);
+		num_subsets = CMath::random(1, 5);
+		min_subset_size = CMath::random(1, 6);
+		base_size = num_labels / num_subsets;
+
+		CRegressionLabels* labels = new CRegressionLabels(num_labels);
+		for (index_t i = 0; i < num_labels; ++i)
+			labels->set_label(i, CMath::random(-10.0, 10.0));
+
+		CTimeSeriesSplitting* splitting =
+		    new CTimeSeriesSplitting(labels, num_subsets);
+
+		splitting->set_min_subset_size(min_subset_size);
+		splitting->build_subsets();
+
+		for (index_t i = 0; i < num_subsets; ++i)
+		{
+			SGVector<index_t> subset = splitting->generate_subset_indices(i);
+			SGVector<index_t> inverse = splitting->generate_subset_inverse(i);
+
+			/* Subset size should be atleat min_subset_size */
+			EXPECT_GE(subset.vlen, splitting->get_min_subset_size());
+
+			/* check the splitting is linear */
+			EXPECT_TRUE(
+			    inverse.vlen % base_size == 0 ||
+			    inverse.vlen == num_labels - min_subset_size);
+		}
+
+		SG_UNREF(splitting);
+	}
+}
+
+TEST(SplittingStrategy, timeseries_subsets_future_leak)
+{
+	index_t num_labels, num_subsets, min_subset_size;
+	index_t runs = 10;
+
+	while (runs-- > 0)
+	{
+		num_labels = CMath::random(50, 150);
+		num_subsets = CMath::random(1, 5);
+		min_subset_size = CMath::random(1, 7);
+
+		CRegressionLabels* labels = new CRegressionLabels(num_labels);
+		for (index_t i = 0; i < num_labels; ++i)
+			labels->set_label(i, CMath::random(-10.0, 10.0));
+
+		CTimeSeriesSplitting* splitting =
+		    new CTimeSeriesSplitting(labels, num_subsets);
+
+		splitting->set_min_subset_size(min_subset_size);
+		splitting->build_subsets();
+
+		for (index_t i = 0; i < num_subsets; ++i)
+		{
+			SGVector<index_t> subset = splitting->generate_subset_indices(i);
+			SGVector<index_t> inverse = splitting->generate_subset_inverse(i);
+
+			/* check future leak into test set */
```

Comment:
this is not what you check. You check that the vector is sorted.
I am not too keen on this test. rather, you should check that *every* element in the training set is smaller than *every* element in the test set.
The manual test mentioned above would check that in a much simpler fashion, so I would just remove this one here

Reply:
Oh, I see. Instead of checking that every element in the training set is smaller than every element in the test set, we can check min in the test set is greater than max in the training set. Will it do the trick?

Reply:
That will absolutely do it.


BUT, as I said, I think the best thing would be to write down an explicit case example that you can solve on paper...

Reply:
Shall I remove both the test with a single manual test using the fixtures?